Sources/SwiftKuerySQLite/SQLiteConnection.swift

```diff
@@ -226,95 +263,110 @@ public class SQLiteConnection: Connection {
+                return
+            }
+            self.runCompletionHandler(SQLitePreparedStatement(statement: unwrappedSqliteStatement), nil, onCompletion: onCompletion)
         }
-        return SQLitePreparedStatement(statement: sqliteStatement!)
     }
 
     /// Execute a prepared statement.
     ///
     /// - Parameter preparedStatement: The prepared statement to execute.
     /// - Parameter onCompletion: The function to be called when the execution has completed.
     public func execute(preparedStatement: PreparedStatement, onCompletion: @escaping ((QueryResult) -> ())) {
-        execute(preparedStatement: preparedStatement, parameters: [Any?](), namedParameters: [String:Any?](), onCompletion: onCompletion)
+        DispatchQueue.global().async {
+            self.execute(preparedStatement: preparedStatement, parameters: [Any?](), namedParameters: [String:Any?](), onCompletion: onCompletion)
+        }
     }
-    
+
     /// Execute a prepared statement with parameters.
     ///
     /// - Parameter preparedStatement: The prepared statement to execute.
     /// - Parameter parameters: An array of the parameters.
     /// - Parameter onCompletion: The function to be called when the execution has completed.
     public func execute(preparedStatement: PreparedStatement, parameters: [Any?], onCompletion: @escaping ((QueryResult) -> ())) {
-        execute(preparedStatement: preparedStatement, parameters: parameters, namedParameters: [String:Any?](), onCompletion: onCompletion)
+        DispatchQueue.global().async {
+            self.execute(preparedStatement: preparedStatement, parameters: parameters, namedParameters: [String:Any?](), onCompletion: onCompletion)
+        }
     }
-    
+
     /// Execute a prepared statement with parameters.
     ///
     /// - Parameter preparedStatement: The prepared statement to execute.
     /// - Parameter parameters: A dictionary of the parameters with parameter names as the keys.
     /// - Parameter onCompletion: The function to be called when the execution has completed.
     public func execute(preparedStatement: PreparedStatement, parameters: [String:Any?], onCompletion: @escaping ((QueryResult) -> ())) {
-        execute(preparedStatement: preparedStatement, parameters: [Any?](), namedParameters: parameters, onCompletion: onCompletion)
+        DispatchQueue.global().async {
+            self.execute(preparedStatement: preparedStatement, parameters: [Any?](), namedParameters: parameters, onCompletion: onCompletion)
+        }
     }
-    
+
     private func execute(preparedStatement: PreparedStatement, parameters: [Any?], namedParameters: [String:Any?], onCompletion: (@escaping (QueryResult) -> ())) {
         guard let statement = preparedStatement as? SQLitePreparedStatement else {
             onCompletion(.error(QueryError.unsupported("Failed to execute unsupported prepared statement")))
```

Comment:
Should this be offloaded?

Reply:
This function is only called from other functions that offload but I believe we should be offloading the completion handler so the current connection (self) will be released for reuse immediately rather than being kept alive for the duration of the handler.

Reply:
Updated to offload.